lti_tool/views.py

```diff
@@ -131,10 +132,48 @@ def parse_lti_session_params(self):
             )
             self.message_launch_data = self.message_launch.get_launch_data()
 
+    def adjust_visibility(self, content):
+        """
+        Check whether given navigation link in ContentCache should be shown in LTI view.
+        This depends on whether LTI link points to single exercise, or whole module/course.
+        """
+        # LTI link points to specific exercise: no navigation allowed
+        if 'exercise_path' in self.lti_scope:
+            return None
+        if not content:
+            return None
+
+        try:
+            if content['type'] == 'exercise':
+                mod = CourseModule.objects.get(pk=content['module_id'])
+                lo = LearningObject.objects.get(pk=content['id'])
```

Comment:
That would be nice, these queries are only needed for building the URL for 'lti-exercise'. But I could not figure out how to build the URL based on the information in the cache. The URL slugs seem to be accessible only through the model class (get_url_kwargs), or is there a way?

Reply:
The first query appears to be totally unused though. I don't know why it is there, that can be removed.

Reply:
Building the URLs without model objects is annoying, indeed. Normally, the URL could be read from the cache, but now the cache entry contains a wrong URL at first. Of course, the URLs can be built with the `reverse()` function, which is what the `get_url()` method does. But it would be slightly more annoying and adds some repetition to the code (against the DRY principle). I guess you might leave the database queries here for now.

> The first query appears to be totally unused though. I don't know why it is there, that can be removed.

Yup, the first if branch does not use the `mod` variable. Please remove that.